sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -1108,7 +1111,7 @@ export class SearchIndexClient {
     readonly endpoint: string;
     getIndex(indexName: string, options?: GetIndexOptions): Promise<SearchIndex>;
     getIndexStatistics(indexName: string, options?: GetIndexStatisticsOptions): Promise<SearchIndexStatistics>;
-    getSearchClient<T>(indexName: string, options?: SearchClientOptions): SearchClient<T>;
+    getSearchClient<T>(indexName: string, options?: getSearchClientOptions): SearchClient<T>;
```

Comment:
@JoshLove-msft Can you confirm if this falls under the category that you mentioned this afternoon in the Service Bus API review i.e when the method is about a service operation, having the naming guideline for the options to use the method name as the prefix applies. But, when the method is about client side object, the guideline does not apply.

cc @bterlson

Reply:
exporting both the old and new types is not elegant. On the other hand, deleting the old type creates a situation like the one here: https://github.com/Azure/azure-sdk-for-js/pull/11506. Perhaps it is best to not update the type for now until the next major release.

Reply:
does this method do a service call? It seems like it does not. If not, then I would think the service operation naming convention wouldn't necessarily apply.

Reply:
This is not a service call, this is locally (and synchronously) creating a client object. So I think it should get an exception.

Reply:
Hi @JoshLove-msft @xirzec to clarify should I get rid of the getSearchClientOptions object, and revert to the old SearchClientOptions type?

Reply:
I would like to revert the change.

Reply:
Hi @JoshLove-msft @xirzec @deyaaeldeen @ramya-rao-a the latest commits should address this.